artiq/coredevice/comm_kernel.py

```diff
@@ -651,6 +655,26 @@ def read_exception_string():
             message = read_exception_string()
             params = [self._read_int64() for _ in range(3)]
 
+            # Add Channel name if any
+            m = re.match(r".+?(channel)\s{(\d+)}", message)
+            if m:
+                format_number = int(m.group(2))
+                channel_number = params[format_number]
+
+                cwd = os.getcwd()
+                path = re.match(r".+?/artiq/\w+", cwd)
+                new_path = path.group(0) + "/device_db.py"
+                device_mgr = DeviceManager(DeviceDB(new_path))
```

Comment:
Oh no.

Reply:
In addition to this code being incredibly clumsy, it will trivially break if ``--device-db`` is passed to the master arguments.

Reply:
> it will trivially break if `--device-db` is passed to the master arguments.

why?